common/mixins.py

```diff
@@ -30,7 +30,8 @@ def clean(self):
         if hasattr(self, 'slugify_field') and hasattr(self, 'slug'):
             self.slug = slugify(getattr(self, self.slugify_field))
 
-            if self.__class__.objects.filter(slug=self.slug).exists():
+            # If pk exists, object exists in db and is being edited by user.
```

Comment:
yes @andela-ooshodi. If the pk does not exist, it means the object is being created newly so the validation checks need to be done on that.